app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ComposeListViewHolder.kt

```diff
@@ -0,0 +1,117 @@
+ * @param featureName [String] representing the name of the feature displaying tabs. Used in telemetry reporting.
+ * @param viewLifecycleOwner [LifecycleOwner] to which this Composable will be tied to.
+ */
+class ComposeListViewHolder(
+    private val interactor: BrowserTrayInteractor,
+    val tabsTrayStore: TabsTrayStore,
+    val selectionHolder: SelectionHolder<TabSessionState>? = null,
+    composeItemView: ComposeView,
+    private val featureName: String,
+    browserStore: BrowserStore,
+    viewLifecycleOwner: LifecycleOwner,
+) : ComposeAbstractTabViewHolder(composeItemView, viewLifecycleOwner), TabItemListener {
+
+    private val viewHolderDelegate =
+        DefaultTabViewHolderDelegate(tabsTrayStore, featureName, browserStore)
+
+    private var delegate: TabsTray.Delegate? = null
+
+    override var tab: TabSessionState? = null
+    private val isSelectedTab = MutableLiveData(false)
+    private val isMultiSelectionSelected = MutableLiveData(false)
+
+    override fun bind(
+        tab: TabSessionState,
+        isSelected: Boolean,
+        styling: TabsTrayStyling,
+        delegate: TabsTray.Delegate
+    ) {
+        this.tab = tab
+        this.delegate = delegate
+        isSelectedTab.value = isSelected
+        bind(tab)
+    }
+
+    override fun updateSelectedTabIndicator(showAsSelected: Boolean) {
+        isSelectedTab.postValue(showAsSelected)
+    }
+
+    override fun showTabIsMultiSelectEnabled(selectedMaskView: View?, isSelected: Boolean) {
+        isMultiSelectionSelected.postValue(isSelected)
+    }
+
+    override fun onCloseClicked(tab: TabSessionState) {
+        delegate?.onTabClosed(tab, featureName)
+    }
+
+    override fun onMediaClicked(tab: TabSessionState) {
+        interactor.onMediaClicked(tab)
+    }
+
+    override fun onClick(tab: TabSessionState) {
+        val holder = selectionHolder
+        if (holder != null) {
+            viewHolderDelegate.handleMultiSelectClick(holder, interactor, tab)
+        } else {
+            viewHolderDelegate.handleTabClick(interactor, tab)
+        }
+    }
+
+    override fun onLongClick(tab: TabSessionState) {
+        val holder = selectionHolder ?: return
+        viewHolderDelegate.handleLongClick(holder, interactor, tab)
+    }
+
+    @Composable
+    override fun Content(tab: TabSessionState) {
+        val isMultiSelectionSelected = isMultiSelectionSelected.observeAsState(false)
+        val isSelectedTab = isSelectedTab.observeAsState(false)
+        TabListItem(
+            tab = tab,
+            isSelected = isSelectedTab.value,
+            multiSelectionEnabled = isSelectModeActive(),
+            multiSelectionSelected = isMultiSelectionSelected.value,
+            onCloseClicked = ::onCloseClicked,
+            onMediaClicked = ::onMediaClicked,
+            onClick = ::onClick,
+            onLongClick = ::onLongClick,
+        )
+    }
+
+    private fun isSelectModeActive() = tabsTrayStore.state.mode is TabsTrayState.Mode.Select
```

Comment:
I think this also has to be made observable.
Saw that if I manually tap some tabs to select and then tap them all to deselect at least one will not show the close ("X") button.
We could probably do 
```
        val isSelectModeActive = tabsTrayStore.observeAsComposableState {
            state -> state.mode is TabsTrayState.Mode.Select
        }.value
```

inside the composable if this is the issue.